src/Symfony/Bridge/Twig/Command/LintCommand.php

```diff
@@ -88,8 +88,39 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $this->excludes = $input->getOption('excludes');
         $this->format = $input->getOption('format') ?? (GithubActionReporter::isGithubActionEnvironment() ? 'github' : 'txt');
 
+        $deprecations = [];
+        if ($showDeprecations) {
+            $prevErrorHandler = set_error_handler(static function ($level, $message, $file, $line) use (&$prevErrorHandler, &$deprecations) {
```

Comment:
I suggest doing this error handler registration inside `validate` (adding the deprecations in the file info), which will avoid having to guess the template file, as it will be available already.

Thus, it will allow to have the proper restoration of the error handler (currently, there is a case where it is not restored)